### DISCUSSION TITLE
[NASA] Samsung heat pump

### DISCUSSION THREAD

**Discussion by Mocky158** [omerfaruk-aran/esphome_samsung_hvac_bus#63] [General]
Good day
Earlier I asked about the connection with the heat pump. Previously, it was a problem to recognize what entity would be for a given value.
I would like to ask if anything has changed since then, if it is possible to recognize the given values?
thank you very much

**Comment by lanwin** [comment]
@Mocky158 I have no idea what you mean.

**Reply by Mocky158** [reply]
It was about the fact that I have a connected heat pump (AE160JNYDGH) and I would need to see the status of the domestic water temperature and be able to change it according to . I would then use the update based on PV surplus.

Earlier when I asked the problem was how to recognize these entities.

**Reply by lanwin** [reply]
The problem here is that we only find out if we try gather data from your device.

**Reply by lanwin** [reply]
There is no official documentation etc where I can find all values.

**Reply by Foxhill67** [reply]
@Mocky158 : may be this helps: https://wiki.myehs.eu/wiki/NASA_Protocol ?
Perhaps this is the message you are looking for: <html>
<body>

0x4237 | VAR_IN_TEMP_WATER_TANK_F | NASA_INDOOR_DHW_CURRENT_TEMP | DHW tank current temperature
-- | -- | -- | --



</body>
</html>

**Reply by lanwin** [reply]
Right. But this is only the current value. The next question is witch value sets the water temp.

**Reply by Foxhill67** [reply]
best guess: <html>
<body>

0x4065 | ENUM_IN_WATER_HEATER_POWER | NASA_DHW_POWER | Water heater power | 0 Off, 1 On
-- | -- | -- | -- | --



</body>
</html> to activate domestic hot water heating? Water will be heated to <html>
<body>

0x4260 | VAR_IN_FSV_3021 |   | DHW Heating mode - Max.
-- | -- | -- | --



</body>
</html> degrees?

**Reply by lanwin** [reply]
Sounds good. Thank you @Foxhill67 

@Mocky158 now its on you to buy the hardware to test that.

**Reply by Mocky158** [reply]
If it is hardware. I have bought and put into operation M5STACK
But for now I only have the basic code I found here.
If you could advise me what I should change, I would be very happy.

![ESP](https://github.com/lanwin/esphome_samsung_ac/assets/90219705/8d983eba-437f-46a4-bd3c-be06a2293c5a)

[yaml.txt](https://github.com/lanwin/esphome_samsung_ac/files/14097780/yaml.txt)

**Reply by lanwin** [reply]
Ok Looks good. Il add basic code for that soon.

**Reply by Mocky158** [reply]
I have already changed a lot of things, but always according to some pattern, if it is possible to guide me just a little, I would try to change the entities in question myself.

**Reply by lanwin** [reply]
@Mocky158 can you please pull and install the current version. I added logging for the values @Foxhill67 provided.

* VAR_IN_TEMP_WATER_TANK_F
* ENUM_IN_WATER_HEATER_POWER
* VAR_IN_FSV_3021
* VAR_IN_FSV_3022
* VAR_IN_FSV_3023

Please post a log with the values you see.

**Reply by Mocky158** [reply]
Here is my log after the update.
If I looked carefully, it shows a water temperature of 44.6, which corresponds to my added thermometer :-) .

[logs_m5-atomic-kotelna_logs.txt](https://github.com/lanwin/esphome_samsung_ac/files/14126212/logs_m5-atomic-kotelna_logs.txt)

**Reply by Mocky158** [reply]
I would like to ask if I can somehow modify the .yaml file so that I can see the temperature values and the heating mode so that I can observe and compare it. When it melts into underfloor heating or domestic water.

**Comment by lanwin** [comment]
Not yet. Currently I try to get the non NASA stuff ready so its useable. Then after I would like to add sensors that can be used for that.

**Reply by Mocky158** [reply]
Well ok will wait for further progress. :-)
